### PR TITLE
refactor and tidy-up sidecar.

### PR DESCRIPTION
**Pull request by raulk**
For better maintainability and shared ownership.

* remove the `dockermanager` package; merge into `docker`.
* fix overusage of term "manager"/"manage".
* shard large source files.

### CODE REVIEW

File: pkg/docker/manager.go [hunk 1]
```diff
@@ -114,38 +114,42 @@ func (dm *Manager) Manage(
...
-				case <-timeout.C:
-					dm.S().Panicw("timed out waiting for container worker to stop", "container", container)
-					return
-				case <-ticker.C:
-					dm.S().Errorw("waiting for container worker to stop", "container", container)
-				}
+	stop := func(containerID string) {
+		wh, ok := managers[containerID]
+		if !ok {
+			return
```
Comment (Stebalien) [inline pkg/docker/manager.go]: :man_facepalming:

File: docs/DEEP_TRACING.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+## Design blueprint: Deep tracing
```
Comment (Stebalien) [inline docs/DEEP_TRACING.md]: Random file?

Comment (coryschwartz) [reply docs/DEEP_TRACING.md]: +1  yeah looks like an accidental git add.

Comment (raulk) [reply docs/DEEP_TRACING.md]: Thanks for noticing, gents!

**Pull request merged**